<launch>
    <arg name="robot_name" value="X1" />

    <include file="$(find noroute_mesh)/launch/converter_service.launch"/>
    <include file="$(find noroute_mesh)/launch/comms_node.launch">
        <arg name="name" value="$(arg robot_name)"/>
        <arg name="id" value="1"/>
    </include>

    <include file="$(find exploration_goal)/launch/robot.launch">
        <arg name="robot_name" value="$(arg robot_name)" />
    </include>

  <arg name="scanPeriod" default="0.1" />

  <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen">
    <param name="lidar" value="VLP-16" /> <!-- options: VLP-16  HDL-32  HDL-64E -->
    <param name="scanPeriod" value="$(arg scanPeriod)" />

    <remap from="/multi_scan_points" to="/X1/points" />
  </node>

  <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen">
    <param name="scanPeriod" value="$(arg scanPeriod)" />
  </node>

  <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen">
  </node>

  <node pkg="lidar_frontier3d" type="frontier" name="frontier_detector">
  </node>

  <include file="$(find noroute_mesh)/launch/converter_service.launch"/>
  <include file="$(find noroute_mesh)/launch/comms_node.launch">
    <arg name="name" value="X1"/>
    <arg name="id" value="1"/>
  </include>

  <node pkg="exploration_goal" type="defecate.py" name="defecation_agent" respawn="true"/>
  <node pkg="exploration_goal" type="get_ur_ass_inside" name="get_ur_ass_inside"/>

  <include file="$(find object_detection)/launch/object_detection.launch"/>
  <include file="$(find artifact_reporter)/launch/artifact_reporter.launch"/>
</launch>